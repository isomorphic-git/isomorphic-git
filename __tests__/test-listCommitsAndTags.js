/* eslint-env node, browser, jasmine */
const { makeFixture } = require('./__helpers__/FixtureFS.js')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')
// @ts-ignore
const snapshots = require('./__snapshots__/test-listCommitsAndTags.js.snap')

const { listCommitsAndTags } = require('isomorphic-git/internal-apis')

describe('listCommitsAndTags', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('listCommitsAndTags', async () => {
    // Setup
    const { fs, gitdir } = await makeFixture('test-listCommitsAndTags')
    // Test
    const commits = await listCommitsAndTags({
      fs,
      gitdir,
      start: ['c60bbbe99e96578105c57c4b3f2b6ebdf863edbc'],
      finish: ['c77052f99c33dbe3d2a120805fcebe9e2194b6f9']
    })
    expect([...commits]).toMatchSnapshot()
  })
})
