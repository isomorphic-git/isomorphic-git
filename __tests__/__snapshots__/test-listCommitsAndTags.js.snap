// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listCommitsAndTags listCommitsAndTags 1`] = `
Array [
  "c60bbbe99e96578105c57c4b3f2b6ebdf863edbc",
  "e05547ea87ea55eff079de295ff56f483e5b4439",
  "ebdedf722a3ec938da3fd53eb74fdea55c48a19d",
  "0518502faba1c63489562641c36a989e0f574d95",
]
`;
